lsusb.c

```diff
@@ -1726,6 +1729,25 @@ static void dump_videocontrol_interface(libusb_device_handle *dev, const unsigne
 	free(term);
 }
 
+static void dump_videocontrol_interrupt_endpoint(const unsigned char *buf)
+{
+	unsigned int j, wMaxTransferSize;
+
+	if (buf[0] < 5)
+		printf("      Warning: Descriptor too short\n");
```

Comment:
Shouldn't you return here?

Reply:
Sounds reasonable. Since I agree with you that it does not make sense to present the decoded descriptor when these basic sanity checks fail, I have updated my code with returns.

This doesn't seem to be done consistently throughout `lsusb`, though. In fact, in most places where a too short descriptor is detected, there is no return. See my examples below.

Would it perhaps be even better to call `dump_junk` on the reported length of the descriptor in these cases, to make it easier for the user to figure out what happened? I would certainly appreciate to see a raw dump, rather than only the error message.


In this example, all three elements of `buf` are printed, even when we know the length is less than 3, which is of course misleading:
```
static void dump_audiocontrol_interface(libusb_device_handle *dev, const unsigned char *buf, int protocol)
{
	enum uac_interface_subtype subtype;

	if (buf[1] != USB_DT_CS_INTERFACE)
		printf("      Warning: Invalid descriptor\n");
	else if (buf[0] < 3)
		printf("      Warning: Descriptor too short\n");
	printf("      AudioControl Interface Descriptor:\n"
	       "        bLength             %5u\n"
	       "        bDescriptorType     %5u\n"
	       "        bDescriptorSubtype  %5u ",
	       buf[0], buf[1], buf[2]);
```

But there are also some examples with a return:
```
static void dump_ccid_device(const unsigned char *buf)
{
	unsigned int us;

	if (buf[0] < 54) {
		printf("      Warning: Descriptor too short\n");
		return;
	}
```